src/test/java/teammates/test/pageobjects/AppPage.java

```diff
@@ -1463,13 +1463,26 @@ void scrollElementToFirstCellAndSendKeys(WebElement spreadsheetElement, String v
         private static final String START_OCCURRED_ATTRIBUTE = "__ajaxStartOccurred__";
 
         /**
-         * Registers `ajaxStart` and `ajaxStop` handlers to track the state of an AJAX request.
+         * Waits for all jQuery AJAX requests to complete.
+         */
+        void waitForAllRequestsComplete() {
+            // `$.active` is a counter for holding the number of active AJAX requests
+            // but is not documented because it is used by JQuery internally.
+            // see https://stackoverflow.com/questions/3148225/jquery-active-function/3148506#3148506
+            waitFor(driver -> (Boolean) ((JavascriptExecutor) driver).executeScript("return $.active === 0"));
+        }
+
+        /**
+         * Waits for all AJAX requests to complete if any is present and registers `ajaxStart` and `ajaxStop` handlers to
+         * track the state of an AJAX request.
          *
          * @throws IllegalStateException if the handlers are already registered in the document
          */
-        void registerHandlers() {
+        void waitForAjaxIfPresentAndRegisterHandlers() {
```

Comment:
I'm not sure which is clearer.
`jQueryAjaxHandler.waitForAjaxIfPresentAndRegisterHandlers` or
`jQueryAjaxHandler.waitForRequestsIfPresentAndRegisterHandlers`.

Or should we just follow:
`jQueryAjaxHandler.waitForNoAjaxAndRegisterHandlers/waitForNoRequestsAndRegisterHandlers`

Reply:
Let's not having `registerHandlers` to take so many responsibilities. I have moved the waiting part to `selectDropdownBy*ValueAndWaitForAjaxRequestComplete`.

Reply:
This was what I originally thought too. But this places the burden on the programmer to ensure that there are no ongoing AJAX requests (which was what made the test fail in the first place) and `registerHandlers` can fail unexpectedly.

Hence, I think there should at least be a check for no ongoing requests and fail the registration. But if it does that, why not go a step further and do it automatically since it's what needs to be done anyway.

Reply:
Ok. Make sense.